File: app/src/main/java/com/example/jacob/myapplication/PeopleListAdapter.java
package com.example.jacob.myapplication;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.TextView;
import com.example.jacob.myapplication.logic.IConversationData;

/**
 * Created by jacob on 21/11/2015.
 */
public class PeopleListAdapter extends BaseAdapter {

    private IConversationData cv;
    private LayoutInflater inflater;

    public PeopleListAdapter(Context context, IConversationData cv) {
        this.cv = cv;
        inflater = (LayoutInflater) context.getSystemService(Context.LAYOUT_INFLATER_SERVICE);
    }

    @Override
    public int getCount() {
        return cv.getParticipants().size();
    }

    @Override
    public Object getItem(int position) {
        return cv.getParticipants().get(position);
    }

    @Override
    public long getItemId(int position) {
        return position;
    }

    @Override
    public View getView(int position, View convertView, ViewGroup parent) {
        convertView = inflater.inflate(R.layout.people_list_item, null);
        TextView tName = (TextView) convertView.findViewById(R.id.people_name);
        TextView tPercentage = (TextView) convertView.findViewById(R.id.people_percentage);
        TextView tWords = (TextView) convertView.findViewById(R.id.words_per_message);
        String name = cv.getParticipants().get(position);
        tName.setText(name);
        String percentage = String.format("%.2f%%", cv.getParticipantShare(name));
        tPercentage.setText(percentage);
        tWords.setText(String.format("%.2f", cv.getWordsAvg(name)));

        return convertView;
    }
}

File: app/src/main/java/me/jacobrr/whatsappanalyzer/PeopleListAdapter.kt
package me.jacobrr.whatsappanalyzer

import android.content.Intent
import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import me.jacobrr.whatsappanalyzer.activities.PersonActivity
import me.jacobrr.whatsappanalyzer.databinding.PeopleListItemBinding
import me.jacobrr.whatsappanalyzer.logic.ConversationData
import me.jacobrr.whatsappanalyzer.logic.IConversationData

/**
 * Created by jacob on 21/11/2015.
 */
class PeopleListAdapter(private val cv: IConversationData, var ready: Boolean = false) :
    RecyclerView.Adapter<PeopleListAdapter.PersonHolder>() {

    class PersonHolder(val view: View) : RecyclerView.ViewHolder(view)

    override fun getItemCount() = cv.participants.size

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): PersonHolder {
        val view = LayoutInflater.from(parent.context)
                .inflate(R.layout.people_list_item, parent, false)
        return PersonHolder(view)
    }

    override fun onBindViewHolder(holder: PersonHolder, position: Int) {
        val binding = PeopleListItemBinding.bind(holder.view)
        val name = cv.participants[position]
        binding.apply {
            peopleName.text = name
            peoplePercentage.text = String.format("%.2f%%", cv.getParticipantShare(name))
            wordsPerMessage.text = String.format("%.2f", cv.getWordsAvg(name))
        }

        Log.d("", "Binding for $name")

        holder.view.setOnClickListener { view ->
            if (ready) {
                Constants.person = (Constants.conversationData as ConversationData).getPersonData(position)
                val intent = Intent(view.context, PersonActivity::class.java)
                view.context.startActivity(intent)
            }

        }
    }

}
